x-pack/plugins/global_search_bar/public/search_syntax/query_utils.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Query } from '@elastic/eui';
+import { FilterValues } from './types';
+
+export const getFieldValueMap = (query: Query) => {
+  const fieldMap = new Map<string, FilterValues>();
+
+  query.ast.clauses.forEach((clause) => {
```

Comment:
Am I right in understanding that this only supports OR logic? Will we need to support AND or more complex logic in the future?

From a UX perspective, I actually find OR less useful than AND in this use case, however, I understand this emulates the Lucene query string behavior. I suspect that it's more common for users to want to use search filters to _narrow_ down results to find a specific object. There is the use case of "I know it had either this tag or this one" but that seems less useful/common than the narrowing case. In general, I suspect users are trying to find a needle in a haystack. That said, I have 0 data to back this up 😄

Reply:
> Am I right in understanding that this only supports OR logic

That's right

> Will we need to support AND or more complex logic in the future?

We might, but that would require changes in both the syntax we use (EUI `Query` syntax does not support `AND` for field terms, we will probably need to use/create our own syntax in that case), and in the SO find API. Also for some filters it doesn't make sense: `type:(dashboard AND map)`.

See  https://github.com/elastic/kibana/issues/74290#issuecomment-727816150 and the reply from alex for more context, but basically, we are not planning this short or mid term.